### PR TITLE
WIFI-991: alb ingress

### PR DESCRIPTION
**Pull request by eugenetaranov-opsfleet**
Upgrade procedure:
- remove load balancers/remove route53 dns records;
- delete nginx-ingress (helm chart or deployment);
- add iam role for alb-ingress;
- install alb-ingress (switch to cloudsdk_testcluster branch in Toolsmith repo, navigate to helmfile/cloud-sdk, enable alb-ingress and run `helmfile -e amazon-testcluster -l group=system apply`);
- run helm upgrade;

### CODE REVIEW

**Review by 4c74356b41 — APPROVED** [review]
lgtm

File: tip-wlan/charts/nginx-ingress-controller/templates/controller-configmap.yaml [hunk 1]
```diff
@@ -7,4 +7,4 @@ metadata:
     {{- include "common.labels" . | nindent 4 }}  
 data:
     external-status-address: {{ .Values.controller.config.externalStatusAddress }}
-    client-max-body-size: {{ .Values.controller.config.clientMaxBodySize }}
\ No newline at end of file
+    client-max-body-size: {{ .Values.controller.config.clientMaxBodySize }}
```
Comment (rahulsharma2020) [inline tip-wlan/charts/nginx-ingress-controller/templates/controller-configmap.yaml]: Why is this file changed?

**Review by leonidlm — APPROVED** [review]
Looks good to me!

File: tip-wlan/charts/opensync-gw-cloud/templates/service.yaml [hunk 2]
```diff
@@ -5,35 +5,49 @@ metadata:
...
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
 spec:
   type: {{ .Values.service.type }}
   ports:
     - port: {{ .Values.service.port1 }}
       targetPort: {{ .Values.service.port1 }}
       protocol: TCP
       name: {{ .Values.service.name1 }}
+      {{- if and .Values.service.nodePort_static (eq .Values.service.type "NodePort") }}
```
Comment (rahulsharma2020) [inline tip-wlan/charts/opensync-gw-cloud/templates/service.yaml]: Please align with other defined variables and use camelcase here: nodePortStatic would be better.

File: tip-wlan/charts/opensync-gw-cloud/values.yaml [hunk 3]
```diff
@@ -77,6 +77,8 @@ service:
   port5: 5005
   name5: debug
   nodePort5: 26
+  annotations: {}
+  nodePort_static: true
```
Comment (rahulsharma2020) [inline tip-wlan/charts/opensync-gw-cloud/values.yaml]: Would be worthwhile to add a comment what nodePort_static means here (why is it defaulted to true) and the intention for it's use.

File: tip-wlan/charts/wlan-cloud-graphql-gw/templates/ingress.yaml [hunk 4]
```diff
@@ -32,6 +32,12 @@ spec:
     - host: {{ .host | quote }}
       http:
         paths:
+        {{- if $.Values.ingress.alb_https_redirect }}
```
Comment (rahulsharma2020) [inline tip-wlan/charts/wlan-cloud-graphql-gw/templates/ingress.yaml]: As @leonidlm pointed earlier, calling this flag alb_https_redirect would make it AWS specific. If this feature of alb is also present in other cloud platforms, perhaps renaming it to `lb_https_redirect` would be better. What do you think?

File: tip-wlan/charts/wlan-cloud-graphql-gw/values.yaml [hunk 5]
```diff
@@ -54,6 +54,7 @@ service:
 
 ingress:
   enabled: true
+  alb_https_redirect: false
```
Comment (rahulsharma2020) [inline tip-wlan/charts/wlan-cloud-graphql-gw/values.yaml]: A comment on what's the intention/purpose of this variable would be good.

File: tip-wlan/charts/wlan-port-forwarding-gateway-service/templates/service.yaml [hunk 6]
```diff
@@ -12,22 +12,22 @@ spec:
...
     {{- if .Values.debug.enabled }}
     - port: {{ .Values.service.port3 }}
       targetPort: {{ .Values.service.port3 }}
       protocol: TCP
       name: {{ .Values.service.name3 }}
-      {{- if eq .Values.service.type "NodePort" }}
+      {{- if and .Values.service.nodePort_static (eq .Values.service.type "NodePort") }}
       nodePort: {{ .Values.global.nodePortPrefixExt | default .Values.nodePortPrefixExt }}{{ .Values.service.nodePort3 }}
       {{- end }}
     {{- end }}
```
Comment (rahulsharma2020) [inline tip-wlan/charts/wlan-port-forwarding-gateway-service/templates/service.yaml:33]: What about the debugPorts (`service.dev.debugport`) in this application? Should we introduce nodePort_static variable for them as well?

File: tip-wlan/charts/wlan-ssc-service/templates/deployment.yaml [hunk 7]
```diff
@@ -216,7 +216,7 @@ spec:
             - name: {{ .Values.service.name3 }}
               containerPort: {{ .Values.service.port3 }}
               protocol: TCP
-            {{- end }}              
+            {{- end }}
```
Comment (rahulsharma2020) [inline tip-wlan/charts/wlan-ssc-service/templates/deployment.yaml:219]: Why is this file changed?

File: tip-wlan/charts/nginx-ingress-controller/templates/controller-configmap.yaml (hunk 1 above)
Comment (eugenetaranov-opsfleet) [reply tip-wlan/charts/nginx-ingress-controller/templates/controller-configmap.yaml]: mm I think editor appended newline .. not important, will revert it

File: tip-wlan/charts/wlan-cloud-graphql-gw/templates/ingress.yaml (hunk 4 above)
Comment (eugenetaranov-opsfleet) [reply tip-wlan/charts/wlan-cloud-graphql-gw/templates/ingress.yaml]: interesting idea, yeah why not, I am pretty sure azure or GCP won't require this but making it a little generic will not hurt

File: tip-wlan/charts/wlan-ssc-service/templates/deployment.yaml (hunk 7 above)
Comment (eugenetaranov-opsfleet) [reply tip-wlan/charts/wlan-ssc-service/templates/deployment.yaml:219]: cleaned up a bunch of whitespaces at the end

File: tip-wlan/charts/wlan-port-forwarding-gateway-service/templates/service.yaml (hunk 6 above)
Comment (eugenetaranov-opsfleet) [reply tip-wlan/charts/wlan-port-forwarding-gateway-service/templates/service.yaml:33]: it is already there @rahulsharma2020, inside `{{- if .Values.debug.enabled }}` if I understand you correctly

Comment (rahulsharma2020) [reply tip-wlan/charts/wlan-port-forwarding-gateway-service/templates/service.yaml:33]: No, I was referring to the debug-ports defined [here](https://github.com/Telecominfraproject/wlan-cloud-helm/blob/master/tip-wlan/charts/common/templates/_ports.tpl#L14) and included in this file. These ports are used by anyone to open a debug session to an AP using Cloud. The debugPorts that you are referring are the one's to debug the application, those are not what I meant.

I don't understand the nodePortStatic role completely, hence not sure if it's applicable to the debugPorts that we opened in the portforwarding application.

Comment (eugenetaranov-opsfleet) [reply tip-wlan/charts/wlan-port-forwarding-gateway-service/templates/service.yaml:33]: ah I see, well the idea behind nodePortStatic is to disable hardcoded nodePort ports as it prevents from deploying multiple cloudsdk on the same cluster (they would conflict on these ports), if nodePortStatic is set to false nodePort ports are chosen by k8s at the deployment time.
It can be left as is assuming that with more than one deployment on the same cluster only one of them would be switched to debug mode.

Comment (rahulsharma2020) [reply tip-wlan/charts/wlan-port-forwarding-gateway-service/templates/service.yaml:33]: I see. 
I am not sure how likely will K8S chooses any ports defined in the values.yaml [here](https://github.com/Telecominfraproject/wlan-cloud-helm/blob/master/tip-wlan/charts/wlan-port-forwarding-gateway-service/values.yaml#L92). They could conflict.

Also, how are you managing opening the firewall (Security Groups for Master/Worker nodes) for opening these ports (and other node ports) to the outside world?

Comment (eugenetaranov-opsfleet) [reply tip-wlan/charts/wlan-port-forwarding-gateway-service/templates/service.yaml:33]: > I am not sure how likely will K8S chooses any ports defined in the values.yaml here. They could conflict.

if it is not a hardcoded value as it is now, it picks any free port from 30000-32767 range.

> Also, how are you managing opening the firewall (Security Groups for Master/Worker nodes) for opening these ports (and other node ports) to the outside world?

alb-ingress controller takes care of adding rule to the worker nodes security group allowing all traffic from LB security group.

**Review by rahulsharma2020 — APPROVED** [review]
